governance/writing-style.md

```diff
@@ -0,0 +1,66 @@
+
+3. **Markdown links.** ```markdown-link-check``` tool has a configuration file called [link-config.json](./link-config.json), that is located in the root of the repo, and it defines a set of rules that the tool following during its execution. It checks for link patterns that it should ignore and the links that should be modified to verify their validity. A good example here is where we place the images - the location for the images is different in the source code and the deployed website, therefore we need to perform validation towards a correct path which may not be the one referenced by the Markdown files in the source code (because those use the reference path that will be valid once the website is built and deployed). If the tool flags a specific link as invalid you need to verify what the error is and modify the link accordingly, on the line and in the file that the tool is referring to. For example, ```FILE: writing-style.md  [✖] https://eco-qube.eu/ → Status: 410``` points to that the respective link in ```writing-style.md``` file is no longer valid and we can confirm it by navigating to the link in the browser. To fix that I would need to update ```https://eco-qube.eu``` with ```https://www.ecoqube.org```, which is its valid alternative. In some cases, for example in some of the Cloudflare-hosted websites, there is stricter protection against web scraping, which causes automated tools like ```markdown-link-check``` to return a ```403``` HTTP error upon accessing respective websites. The same error will also occur if respective websites are accessed via tools like ```cUrl``` or ```wget```. The alternative solution for tools like ```cUrl``` and ```wget``` is to use open source tools that fully simulate a browser, but this is not something that is currently supported by the ```markdown-link-check tool```. Therefore, **if the tool flags a URL as invalid, but you're able to access it via browser**, please add a ```<!-- markdown-link-check-disable-next-line -->``` flag right before the line with the link to disable checking of that link by the tool.
```

Comment:
```suggestion
3. **Markdown links.** ```markdown-link-check``` tool has a configuration file called [link-config.json](../link-config.json), that is located in the root of the repo, and it defines a set of rules that the tool following during its execution. It checks for link patterns that it should ignore and the links that should be modified to verify their validity. A good example here is where we place the images - the location for the images is different in the source code and the deployed website, therefore we need to perform validation towards a correct path which may not be the one referenced by the Markdown files in the source code (because those use the reference path that will be valid once the website is built and deployed). If the tool flags a specific link as invalid you need to verify what the error is and modify the link accordingly, on the line and in the file that the tool is referring to. For example, ```FILE: writing-style.md  [✖] https://eco-qube.eu/ → Status: 410``` points to that the respective link in ```writing-style.md``` file is no longer valid and we can confirm it by navigating to the link in the browser. To fix that I would need to update ```https://eco-qube.eu``` with ```https://www.ecoqube.org```, which is its valid alternative. In some cases, for example in some of the Cloudflare-hosted websites, there is stricter protection against web scraping, which causes automated tools like ```markdown-link-check``` to return a ```403``` HTTP error upon accessing respective websites. The same error will also occur if respective websites are accessed via tools like ```cUrl``` or ```wget```. The alternative solution for tools like ```cUrl``` and ```wget``` is to use open source tools that fully simulate a browser, but this is not something that is currently supported by the ```markdown-link-check tool```. Therefore, **if the tool flags a URL as invalid, but you're able to access it via browser**, please add a ```<!-- markdown-link-check-disable-next-line -->``` flag right before the line with the link to disable checking of that link by the tool.
```